some well needed comments and cosmetic changes to the pages

thedekel opened the pull request:
### changes include
- Org page now shows last 10 events for the org
- Format assistance for time-inputs on create-event page
- proper field validation and back-end validation for kiosk-mode attendance attempts
- replaced deprecated mongoose functions with appropriate substitute functions
- comments and documentation on .js files
- .js files now validate with jslint
- updated readme with more info and introducing a new validation/documentation policy

thedekel committed:
kiosk error-checking, create-event time-format fix

thedekel committed:
incorrect port used for testing

thedekel committed:
started cosmetic changes and documentation

thedekel committed:
minor cosmetic changes to kiosk mode page

thedekel committed:
removed discount (replaced with 'marked')

thedekel committed:
added Events list to /orgs/:org

thedekel committed:
Commented and verified jslint on all .js files

thedekel committed:
jslit -verification for auth.js

thedekel committed:
fixed reversed sort directions on sort()

thedekel committed:
cosmetic changes to code and pages, updated readme

thedekel committed:
changed git repo url from thelastnode to cc-wit

thedekel committed:
fixed error with unguesting not having "event.org" in scope

thedekel committed:
added nohup.out to gitignore

thedekel committed:
fixed messy db error

thedekel committed:
almost working migration into express 3.0

Stupid errors due to everyauth refusing to work with express

thedekel committed:
fixed issue with everyauth, fixed styles

thedekel committed:
Merge branch 'migration2'

Conflicts:
	.gitignore
	app.js
	lib/auth.js
	package.json
	routes/events.js
	routes/orgs.js
	views/create-event.jade
	views/create-org.jade
	views/event.jade
	views/events.jade
	views/index.jade
	views/kiosk.jade
	views/layout.jade
	views/login.jade
	views/org.jade
	views/orgs.jade
	views/recover.jade
	views/register.jade
	views/reset-password.jade

thedekel committed:
disabled message-flashing, updated auth

thedekel committed:
fixed minor bug with events and orgs view

thedekel committed:
updated for working with nodejitsu

thedekel committed:
IT WAS THE MONGOSTORE ALL ALONG!

thedekel committed:
heroku doesn't like an unspecified npm version

thedekel committed:
don't forget about those silly, silly commas now

thedekel committed:
except now, this is agood time to forget that one comma

thedekel committed:
reversed the order of orgs and event listing (fixed)

thedekel committed:
fixed 403 on create-org, fixed update-org

also, lists are now sorted correctly.

thedekel committed:
started admin panel

thedekel committed:
i need mongo access dammit

thedekel committed:
stuff is hammers

thedekel committed:
take recovery out, it's broken and shouldn't be in coffee script

Magicjarvis commented on app.js:
```diff
@@ -19,13 +32,17 @@ everyauth.password
-            next: req.query.next
+            next: req.query.next,
+            req: req,
+            user:req.user
```
spaces are missing here and on line 37.

thelastnode commented on app.js:
```diff
@@ -19,13 +32,17 @@ everyauth.password
     .authenticate(auth.authenticate)
     .loginLocals(function(req, res) {
```
Don't use TODOs, use Github issues
On Aug 26, 2012 9:27 PM, "Jarvis Johnson" notifications@github.com wrote:

> In app.js:
> 
> > @@ -19,13 +32,17 @@ everyauth.password
> >      .authenticate(auth.authenticate)
> >      .loginLocals(function(req, res) {
> 
> add //TODO(thedekel): Figure out the difference between
> loginLocals/registerLocals
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/gt-webdev/attendance/pull/153/files#r1461615.

Magicjarvis commented on conf.js:
```diff
@@ -14,3 +14,15 @@ module.exports = {
 };
+var urire = /mongodb:\/\/(\w+?):(\w+?)@([\w\.]+?):(\d+?)\/(\w+)/,
+    res;
+if (urire.test(module.exports.mongo.uri)){
```
if (...) {

Magicjarvis commented on app.js:
```diff
@@ -40,78 +57,80 @@ everyauth.password
-
-// Configuration
+everyauth.everymodule.findUserById( function(req, userId, callback){
+  User.findOne({_id: userId}, function(err, user) {
```
Do we want functions to look like this or not? I do. Let's make it uniform. Line 60 has a different style than line 61.

thedekel replied:
All functions should now look like this.
function(params) {
}

Magicjarvis commented on public/stylesheets/style.css:
```diff
@@ -57,51 +64,51 @@ h2 small a:hover {
 @media (min-width: 980px) {
 
-/* Fixes a bug where the collapse div has overflow:hidden, even after the window is expanded, causing any nav bar dropdowns to be hidden */
+  /* Fixes a bug where the collapse div has overflow:hidden, even after the window is expanded, causing any nav bar dropdowns to be hidden */
```
Exceeds 80 characters.

Magicjarvis commented on routes/events.js:
```diff
@@ -4,558 +4,681 @@ var models = require('../lib/models');
+         * models to accomplish this. However, this is kept for legacy
+         * support purposes - NOTE: if the database is ever restarted,
+         * this functionality should be safe to remove! */
+        //further note: production database has been successfully migrated
```
There's probably a better way to document this. Also, we should probably stick to one style of comment within functions.

Magicjarvis commented on routes/events.js:
```diff
@@ -4,558 +4,681 @@ var models = require('../lib/models');
+        var page = req.query.page || 0;
+        var limit = 10;
+        //find all events, and fetch the corresponding org (instead of id)
+        var q = models.Event.find().populate('org');
```
q is not a descriptive variable name.

thedekel replied:
Usually q is my temporary-variable name of choice (because it's pretty easy to hit), but at this point, it was `q` before I touched the code. It stands for Query.

Magicjarvis commented on routes/index.js:
```diff
@@ -30,8 +31,8 @@ exports.registerOn = function(app) {
     app.put('/events/:id/kiosk', events.guest);
 
-    app.get('/recover', users.recover);
+    /*app.get('/recover', users.recover);
```
If it's commented out like this, it should probably be removed.

thedekel replied:
this was just a temporary fix to get rid of the password recovery flow which was breaking the app, will be returned later.

Magicjarvis commented on routes/orgs.js:
```diff
@@ -1,133 +1,204 @@
-            orgs: orgs,
-        });
+  async.waterfall([
+                  //first, find all the orgs, sort by name (ascending)
```
What's up with the indentation?

thedekel replied:
Vim's autoindent-derps. doesn't seem to like functions within lists within functions, I'll go over these manually.

thedekel commented:
Keep this open while I make the changes on my version, this request will update when I'm done and I'll comment on everything I fixed.

thedekel committed:
cleaned up some formatting, reploced coffeescript

thedekel committed:
fiexed silly error with hitting an org that doesn't exist

thedekel committed:
fixed poorly formatted e-mail recovery e-mails

thedekel committed:
fixed issues with generating password-recovery views

also fixed the login screen and prevented password recovery
when a user is logged in.

thedekel committed:
returned csrf value to normal

thedekel commented:
I believe all comments were addressed, feel free to re-review

I've also included some fixes and modifications based on the last day of using the app.

thedekel committed:
Merge branch 'master' into admin

thedekel committed:
implemented admin panel at /admin

Magicjarvis commented:
Merging this. We should have much smaller pull requests in the future (though I understand the circumstances of this one)
